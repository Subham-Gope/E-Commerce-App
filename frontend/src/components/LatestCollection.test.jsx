import { describe, test, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import LatestCollection from "./LatestCollection";
import ShopContext from "../context/ShopContext";
import { MemoryRouter } from "react-router-dom";

// mock products
const latestProducts = [
  { _id: "1", image: ["image1"], price: 5, name: "product1" },
  { _id: "2", image: ["image2"], price: 10, name: "product2" },
  { _id: "3", image: ["image3"], price: 15, name: "product3" },
  { _id: "4", image: ["image4"], price: 20, name: "product4" },
  { _id: "5", image: ["image5"], price: 25, name: "product5" },
  { _id: "6", image: ["image6"], price: 30, name: "product6" },
  { _id: "7", image: ["image7"], price: 35, name: "product7" },
  { _id: "8", image: ["image8"], price: 40, name: "product8" },
  { _id: "9", image: ["image9"], price: 45, name: "product9" },
  { _id: "10", image: ["image10"], price: 50, name: "product10" },
  { _id: "11", image: ["image11"], price: 55, name: "product11" },
];

// Reusable render function
const renderWithContext = (ui, value) => {
  return render(
    <MemoryRouter>
      <ShopContext.Provider value={value}>{ui}</ShopContext.Provider>
    </MemoryRouter>
  );
};

describe("Testing the LatestCollection.jsx file", () => {
  test("displays title and 10 products", () => {
    // rendering and setting up the snap shot matching
    const { container } = renderWithContext(<LatestCollection />, {
      products: latestProducts,
    });
    expect(container).toMatchSnapshot();

    // checking for the title and description
    const title = screen.getByText(/latest collect/i);
    expect(title).toBeInTheDocument();
    const description = screen.getByText(/discover our latest collection/i);
    expect(description).toBeInTheDocument();

    // checking the products
    const products = screen.queryAllByText(/product/i);
    expect(products).toHaveLength(10);

    // 11th product shall not be displayed
    const product11 = screen.queryByText(/product 11/i);
    expect(product11).toBeNull();
  });
});
