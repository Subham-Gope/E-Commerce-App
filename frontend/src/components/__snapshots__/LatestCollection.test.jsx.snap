// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the LatestCollection.jsx file > displays title and 10 products 1`] = `
<div>
  <div
    class="my-10"
  >
    <div
      class="text-center py-8 text-3xl"
    >
      <div
        class="inline-flex gap-2 items-center mb-3"
      >
        <p
          class="text-gray-500"
        >
          LATEST
           
          <span
            class="text-gray-700 font-medium"
          >
            COLLECTION
          </span>
        </p>
        <p
          class="w-8 sm:w-12 h-[1px] sm:h-[2px] bg-gray-700"
        />
      </div>
      <p
        class="w-3/4 m-auto text-xs sm:text-sm md:text-base text-gray-600"
      >
        Discover Our Latest Collection: Trendy Styles and Must-Have Pieces Just for You!
      </p>
    </div>
    <div
      class="grid grid-cols-2 sm:grid-cols-3 md:grid-cols-4 lg:grid-cols-5 gap-4 gap-y-6"
    >
      <a
        class="text-gray-700 cursor-pointer"
        data-discover="true"
        href="/product/1"
      >
        <div
          class="overflow-hidden"
        >
          <img
            alt="product image"
            class="hover:scale-110 transition ease-in-out"
            src="image1"
          />
        </div>
        <p
          class="pt-3 pb-1 text-sm"
        >
          product1
        </p>
        <p
          class="text-sm font-medium"
        >
          5
        </p>
      </a>
      <a
        class="text-gray-700 cursor-pointer"
        data-discover="true"
        href="/product/2"
      >
        <div
          class="overflow-hidden"
        >
          <img
            alt="product image"
            class="hover:scale-110 transition ease-in-out"
            src="image2"
          />
        </div>
        <p
          class="pt-3 pb-1 text-sm"
        >
          product2
        </p>
        <p
          class="text-sm font-medium"
        >
          10
        </p>
      </a>
      <a
        class="text-gray-700 cursor-pointer"
        data-discover="true"
        href="/product/3"
      >
        <div
          class="overflow-hidden"
        >
          <img
            alt="product image"
            class="hover:scale-110 transition ease-in-out"
            src="image3"
          />
        </div>
        <p
          class="pt-3 pb-1 text-sm"
        >
          product3
        </p>
        <p
          class="text-sm font-medium"
        >
          15
        </p>
      </a>
      <a
        class="text-gray-700 cursor-pointer"
        data-discover="true"
        href="/product/4"
      >
        <div
          class="overflow-hidden"
        >
          <img
            alt="product image"
            class="hover:scale-110 transition ease-in-out"
            src="image4"
          />
        </div>
        <p
          class="pt-3 pb-1 text-sm"
        >
          product4
        </p>
        <p
          class="text-sm font-medium"
        >
          20
        </p>
      </a>
      <a
        class="text-gray-700 cursor-pointer"
        data-discover="true"
        href="/product/5"
      >
        <div
          class="overflow-hidden"
        >
          <img
            alt="product image"
            class="hover:scale-110 transition ease-in-out"
            src="image5"
          />
        </div>
        <p
          class="pt-3 pb-1 text-sm"
        >
          product5
        </p>
        <p
          class="text-sm font-medium"
        >
          25
        </p>
      </a>
      <a
        class="text-gray-700 cursor-pointer"
        data-discover="true"
        href="/product/6"
      >
        <div
          class="overflow-hidden"
        >
          <img
            alt="product image"
            class="hover:scale-110 transition ease-in-out"
            src="image6"
          />
        </div>
        <p
          class="pt-3 pb-1 text-sm"
        >
          product6
        </p>
        <p
          class="text-sm font-medium"
        >
          30
        </p>
      </a>
      <a
        class="text-gray-700 cursor-pointer"
        data-discover="true"
        href="/product/7"
      >
        <div
          class="overflow-hidden"
        >
          <img
            alt="product image"
            class="hover:scale-110 transition ease-in-out"
            src="image7"
          />
        </div>
        <p
          class="pt-3 pb-1 text-sm"
        >
          product7
        </p>
        <p
          class="text-sm font-medium"
        >
          35
        </p>
      </a>
      <a
        class="text-gray-700 cursor-pointer"
        data-discover="true"
        href="/product/8"
      >
        <div
          class="overflow-hidden"
        >
          <img
            alt="product image"
            class="hover:scale-110 transition ease-in-out"
            src="image8"
          />
        </div>
        <p
          class="pt-3 pb-1 text-sm"
        >
          product8
        </p>
        <p
          class="text-sm font-medium"
        >
          40
        </p>
      </a>
      <a
        class="text-gray-700 cursor-pointer"
        data-discover="true"
        href="/product/9"
      >
        <div
          class="overflow-hidden"
        >
          <img
            alt="product image"
            class="hover:scale-110 transition ease-in-out"
            src="image9"
          />
        </div>
        <p
          class="pt-3 pb-1 text-sm"
        >
          product9
        </p>
        <p
          class="text-sm font-medium"
        >
          45
        </p>
      </a>
      <a
        class="text-gray-700 cursor-pointer"
        data-discover="true"
        href="/product/10"
      >
        <div
          class="overflow-hidden"
        >
          <img
            alt="product image"
            class="hover:scale-110 transition ease-in-out"
            src="image10"
          />
        </div>
        <p
          class="pt-3 pb-1 text-sm"
        >
          product10
        </p>
        <p
          class="text-sm font-medium"
        >
          50
        </p>
      </a>
    </div>
  </div>
</div>
`;
